test-utils/runtime/src/lib.rs

```diff
@@ -284,349 +156,153 @@ pub type Block = sp_runtime::generic::Block<Header, Extrinsic>;
+		Ok(())
 	}
-	fn name<P: 'static>() -> Option<&'static str> {
-		let type_id = sp_std::any::TypeId::of::<P>();
-		if type_id == sp_std::any::TypeId::of::<system::Pallet<Runtime>>() {
-			return Some("System")
-		}
-		if type_id == sp_std::any::TypeId::of::<pallet_timestamp::Pallet<Runtime>>() {
-			return Some("Timestamp")
-		}
-		if type_id == sp_std::any::TypeId::of::<pallet_babe::Pallet<Runtime>>() {
-			return Some("Babe")
-		}
 
-		None
-	}
-	fn module_name<P: 'static>() -> Option<&'static str> {
-		let type_id = sp_std::any::TypeId::of::<P>();
-		if type_id == sp_std::any::TypeId::of::<system::Pallet<Runtime>>() {
-			return Some("system")
-		}
-		if type_id == sp_std::any::TypeId::of::<pallet_timestamp::Pallet<Runtime>>() {
-			return Some("pallet_timestamp")
-		}
-		if type_id == sp_std::any::TypeId::of::<pallet_babe::Pallet<Runtime>>() {
-			return Some("pallet_babe")
+	fn validate(
+		&self,
+		_who: &Self::AccountId,
+		call: &Self::Call,
+		_info: &DispatchInfoOf<Self::Call>,
+		_len: usize,
+	) -> TransactionValidity {
+		log::trace!(target: LOG_TARGET, "validate");
+		if let RuntimeCall::SubstrateTest(ref substrate_test_call) = call {
+			return substrate_test_pallet::validate_runtime_call(substrate_test_call)
 		}
-
-		None
+		Ok(ValidTransaction { provides: vec![vec![0u8]], ..Default::default() })
 	}
-	fn crate_version<P: 'static>() -> Option<CrateVersion> {
-		use frame_support::traits::PalletInfoAccess as _;
-		let type_id = sp_std::any::TypeId::of::<P>();
-		if type_id == sp_std::any::TypeId::of::<system::Pallet<Runtime>>() {
-			return Some(system::Pallet::<Runtime>::crate_version())
-		}
-		if type_id == sp_std::any::TypeId::of::<pallet_timestamp::Pallet<Runtime>>() {
-			return Some(pallet_timestamp::Pallet::<Runtime>::crate_version())
-		}
-		if type_id == sp_std::any::TypeId::of::<pallet_babe::Pallet<Runtime>>() {
-			return Some(pallet_babe::Pallet::<Runtime>::crate_version())
-		}
 
-		None
+	fn pre_dispatch(
+		self,
+		who: &Self::AccountId,
+		call: &Self::Call,
+		info: &sp_runtime::traits::DispatchInfoOf<Self::Call>,
+		len: usize,
+	) -> Result<Self::Pre, TransactionValidityError> {
+		self.validate(who, call, info, len).map(|_| SignedExtraDummy {})
 	}
 }
 
-parameter_types! {
-	pub const DbWeight: RuntimeDbWeight = RuntimeDbWeight {
-		read: 100,
-		write: 1000,
-	};
-	pub RuntimeBlockLength: BlockLength =
-		BlockLength::max(4 * 1024 * 1024);
-	pub RuntimeBlockWeights: BlockWeights =
-		BlockWeights::with_sensible_defaults(Weight::from_parts(4 * 1024 * 1024, 0), Perbill::from_percent(75));
-}
-
-impl From<frame_system::Call<Runtime>> for Extrinsic {
-	fn from(_: frame_system::Call<Runtime>) -> Self {
-		unimplemented!("Not required in tests!")
+construct_runtime!(
+	pub enum Runtime where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = Extrinsic
+	{
+		System: frame_system,
+		Babe: pallet_babe,
+		SubstrateTest: substrate_test_pallet::pallet,
```

Comment:
```suggestion
		SubstrateTest: substrate_test_pallet::pallet,
		Balances: pallet_balances,
		Sudo: pallet_sudo,
		RootTesting: pallet_root_testing,
```

Then we should have all the functionality we need